test/browser/call-device.js

```diff
@@ -0,0 +1,154 @@
+        assert(res.some(dev => dev.clientId === user.clientId));
+    });
+
+    it('should get audio and video stats', async () => {
+        const res = await client.getAudioVideoStats();
+        assert(res);
+    });    
+    
+    it('should get local audio and video stream', async () => {
+        stream = await client.getLocalAudioVideoStream();
+        assert(stream);
+    });    
+    it('should set local audio and video stream', async () => {
+        await client.setAudioVideoStream(call.callId, stream);
+    });
+
+    it('should toggle remote audio OFF', async () => {
+        await Promise.all([
+            client.toggleRemoteAudio(call.callId),
+            expectEvents(client, [{
+                type: 'callStatus',
+                predicate: evt => evt.reason === 'remoteStreamUpdated' && evt.call.callId === call.callId && evt.call.remoteAudioDisabled
+            }])
+        ]);
+    });
+
+    it('should toggle remote audio ON', async () => {
+        await Promise.all([
+            client.toggleRemoteAudio(call.callId),
+            expectEvents(client, [{
+                type: 'callStatus',
+                predicate: evt => evt.reason === 'remoteStreamUpdated' && evt.call.callId === call.callId && !evt.call.remoteAudioDisabled
+            }])
+        ]);
+    });
+
+    it('should get remote streams', async () => {
+        const res = await client.getRemoteStreams(call.callId);
+        assert(res);
+    });
+
+    it('should toggle remote video OFF', async () => {
+        await Promise.all([
+            client.toggleRemoteVideo(call.callId),
+            expectEvents(client, [{
+                type: 'callStatus',
+                predicate: evt => evt.reason === 'sdpConnected' && evt.call.callId === call.callId && evt.call.remoteVideoDisabled
+            }])
+        ]);
+    });
+
+    it('should toggle remote video ON', async () => {
+        await Promise.all([
+            client.toggleRemoteVideo(call.callId),
+            expectEvents(client, [{
+                type: 'callStatus',
+                predicate: evt => evt.reason === 'sdpConnected' && evt.call.callId === call.callId && !evt.call.remoteVideoDisabled
+            }])
+        ]);
+    });
+
+    it('should set media devices', async () => {
+        const  mediaDevices = await navigator.mediaDevices.enumerateDevices();
+        const videoInputDevice = mediaDevices.find(d => d.kind === 'videoinput');
+        const audioInputDevice = mediaDevices.find(d => d.kind === 'audioinput');
+        const audioOutputDevice = mediaDevices.find(d => d.kind === 'audiooutput');
+        let devices = {
+            recording: audioInputDevice.deviceId,
+            playback: audioOutputDevice.deviceId,
+            video: videoInputDevice.deviceId,
+            ringiing: audioOutputDevice.deviceId
+        }
+        const r = await Promise.all([
+            client.setMediaDevices(devices),
+            expectEvents(client, [{
```

Comment:
Here we aren't really testing anything regarding devices. But again that's difficult. For now that's fine.